src/core/base.jl

```diff
@@ -6,16 +6,58 @@ export
     run_generic_model, build_generic_model, solve_generic_model
 
 ""
-@compat abstract type AbstractPowerModel end
-
-""
 @compat abstract type AbstractPowerFormulation end
 
 ""
 @compat abstract type AbstractConicPowerFormulation <: AbstractPowerFormulation end
 
-""
-type GenericPowerModel{T<:AbstractPowerFormulation} <: AbstractPowerModel
+"""
+```
+type GenericPowerModel{T<:AbstractPowerFormulation}
+    model::JuMP.Model
+    data::Dict{String,Any} 
+    setting::Dict{String,Any}
+    solution::Dict{String,Any}
+    ref::Dict{Symbol,Any} # reference data
+end
+```
+where
+
+* `data` is the original data, usually from reading in a `.json` or `.m` (patpower) file
+* `setting` usually looks something like `Dict("output" => Dict("line_flows" => true))`, and
+* `ref` is a `Dict` with the following keys:
```

Comment:
> If you prefer, I can change it to move the current description of ref to build_ref(), and use your comment in place of the description in GenericPowerModel.

I think this is the best plan for now.  An in-depth explanation of how to utilize ref to build a model is best left  for an in-depth tutorial.  It's really only there as a convenience, users can work from the data directly, if they wish.